x-pack/plugins/observability_onboarding/public/assets/standalone_agent_setup.sh

```diff
@@ -2,27 +2,121 @@
 
 API_KEY_ENCODED=$1
 API_ENDPOINT=$2
+ELASTIC_AGENT_VERSION=$3
+AUTO_DOWNLOAD_CONFIG=$4
 
 updateStepProgress() {
-  echo "  GET $API_ENDPOINT/step/$1?status=$2"
+  local STEPNAME="$1"
+  local STATUS="$2" # "incomplete" | "complete" | "disabled" | "loading" | "warning" | "danger" | "current"
   curl --request GET \
-    --url "$API_ENDPOINT/step/$1?status=$2 2023-05-24" \
-    --header "Authorization: ApiKey $API_KEY_ENCODED" \
+    --url "${API_ENDPOINT}/custom_logs/step/${STEPNAME}?status=${STATUS}" \
+    --header "Authorization: ApiKey ${API_KEY_ENCODED}" \
     --header "Content-Type: application/json" \
-    --header "kbn-xsrf: true"
-  echo ""
+    --header "kbn-xsrf: true" \
+    --output /dev/null \
+    --no-progress-meter
 }
 
-echo "Downloading Elastic Agent"
-# https://www.elastic.co/guide/en/fleet/8.7/install-standalone-elastic-agent.html
-curl -L -O https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-8.7.1-linux-x86_64.tar.gz
-updateStepProgress "ea-download" "success"
+echo "Downloading Elastic Agent archive"
+updateStepProgress "ea-download" "loading"
+ELASTIC_AGENT_DOWNLOAD_URL="https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-${ELASTIC_AGENT_VERSION}-linux-x86_64.tar.gz"
```

Comment:
Is the plan to have a script for each OS or is the idea that this becomes more a template that is rendered before downloading? Asking because of the hardcoding of the url -> linux specifics.

Reply:
Follow up for this topic: https://github.com/elastic/kibana/issues/159382